### PR TITLE
Set up the reducers and app structure

### PR DESCRIPTION
**Pull request by JOdiemo**
Understand the concept of state management.
Use store, actions, and reducers in React.

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @JOdiemo, kindly add a descriptive and professional readme to your project 👍🏿

File: README.md [hunk 1]
```diff
@@ -1,70 +1,180 @@
...
+
+I would like to acknowledge the Microverse to give me this opportunity.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (prolajumokeoni) [inline README.md:181]: - [ ] Kindly fix the link to your MIT page, it is currently broken 👍🏿

File: src/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { configureStore } from '@reduxjs/toolkit';
+import bookReducer from './books/booksSlice';
+
+const store = configureStore({
+  reducer: bookReducer,
+});
```
Comment (prolajumokeoni) [inline src/redux/store.js]: - [ ] Please import categories from src/redux/categories and add categories to `configureStore` method, see example below 👍🏿 

```suggestion

const store = configureStore({
  `reducer: { books: booksReducer, categories: categoriesReducer },`
});
```

File: src/redux/categories/categoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const categoriesSlice = () => (
+  <div>
+    Under Construction;
+  </div>
+);
+
+export default categoriesSlice;
```
Comment (prolajumokeoni) [inline src/redux/categories/categoriesSlice.js]: - [ ] Please update the content of this following the requirements .
- [ ]  In this file,  you should:
`Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)`

- Please check the requirements 👍🏿

File: src/redux/books/booksSlice.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: () => {
+
+    },
+    removeBook: () => {
+
```
Comment (prolajumokeoni) [inline src/redux/books/booksSlice.js]: - [ ] Please create the  2 reducers, `addBook` and `removeBook` here, this is part of the requirements, See suggestion 👍🏿 
```suggestion
  ```javascript
 addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
```

```

File: src/redux/books/booksSlice.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    },
+  },
+});
+
```
Comment (prolajumokeoni) [inline src/redux/books/booksSlice.js]: Please export `addBook` and `removeBook` as actions this is part of the requirements 👍🏿 
```suggestion
export const { addBook, removeBook } = booksSlice.actions
```

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @JOdiemo 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - Installed and used  Redux Toolkit 👍🏿 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — APPROVED** [review]
Hi @JOdiemo 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**